utils.go

```diff
@@ -47,6 +49,7 @@ func flipPathBit(data []byte, position int) {
 	data[position/8] = byte(n)
 }
 
+// countSetBits counts the number of bits set in the data provided (ie the number of 1s)
 func countSetBits(data []byte) int {
```

Comment:
This is a micro optimization but sharing for fun. YOu can potentially avoid using `getPathBit` by using the `Kernighan’s` method.

Didn't test it but here's a code snippet chatgpt gave me:

```go
func countSetBits(data []byte) int {
	count := 0
	for _, b := range data {
		// Kernighan’s Method of counting set bits in a byte
		for b != 0 {
			b = b & (b - 1) // unset the rightmost set bit
			count++
		}
	}
	return count
}
```